fix #13916, legendre polynomail for negative n

vivonk opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->
## Change in code for negative n in Legendre polynomial

#### Fixes to issues #13916 
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234". See
https://github.com/blog/1506-closing-issues-via-pull-requests .--

vivonk committed:
fix #13916, legendre polynomail for negative n

normalhuman commented on sympy/functions/special/polynomials.py:
```diff
@@ -799,8 +799,8 @@ def eval(cls, n, x):
-                raise ValueError(
-                    "The index n must be nonnegative integer (got %r)" % n)
+                n = -n - S.One
+                return cls._eval_at_order(n, x)
```
Instead of repeating the same return statement, put it simpler:
```
if n.is_negative:
    n = -n - S.One
return cls._eval_at_order(n, x)
 ```
Having fewer return statements is better, the code gets easier to track.

normalhuman commented:
The text of pull request (which you can edit) should describe the change that was made. For example, "instead of raising an error when n is negative, replace n by -n-1 which is a convention (link) with Legendre polynomials",

vivonk committed:
fix #13916, legendre polynomial for negative n

Instead of giving exception for negative n in Legendre Polynomial, replacing n to -n-1. This is a convention(http://functions.wolfram.com/Polynomials/LegendreP/02/) with Legendre polynomials

normalhuman commented:
I forgot: there should be a test demonstrating that the fix works. Within [this function](https://github.com/sympy/sympy/blob/master/sympy/functions/special/tests/test_spec_polynomials.py#L93), add a test such as 
```
assert legendre(-4, x) == ((5*x**3 - 3*x)/2).expand()
```
to show that negative orders actually work now.

normalhuman reviewed (changes requested):
Good solution, but needs  a test to go with it

vivonk committed:
fix #13916, legendre polynomail for negative n, test cases added

Instead of giving exception for negative n in Legendre Polynomial, replacing n to -n-1. This is a convention(http://functions.wolfram.com/Polynomials/LegendreP/02/) with Legendre polynomials. Test cases are added too in test_spec_polynomials.py' to show that negative n works fine.

vivonk committed:
fix #13916, legendre polynomail for negative n, test cases added

Instead of giving exception for negative n in Legendre Polynomial, replacing n to -n-1. This is a convention(http://functions.wolfram.com/Polynomials/LegendreP/02/) with Legendre polynomials. Test cases are added too in test_spec_polynomials.py' to show that negative n works fine.

vivonk committed:
Delete workspace.xml

vivonk commented:
Added test cases, but there is the problem that one extra file has been commit which is IDE file.

vivonk committed:
fixed local repo

vivonk committed:
fixed local repo
